bayes_opt/target_space.py

```diff
@@ -237,17 +331,15 @@ def register(self, params, target, constraint_value=None):
                                                       [constraint_value]])
 
     def probe(self, params):
-        """
-        Evaluates a single point x, to obtain the value y and then registers the
-        point and its evaluation.
+        """Evaluate the target function on a point and register the result.
 
         Notes
         -----
-        If x has been previously seen returns a cached value of y.
+        If `params` has been previously seen returns a cached value of `y`.
```

Comment:
I'm not sure I see this functionality in this function. I think its calculating and returning target every time.

Reply:
You're right, this was removed in #372. Should we adjust the behaviour here to return the cached value or change the docstring to reflect the current code (where this would raise an error).

Reply:
In either case, we should probably check whether a point is already registered before evaluating the target function potentially unnecessarily?

Reply:
I changed the behaviour to be in line with the docstring again.